src/main/java/app/model/Grid/Grid.java

```diff
@@ -0,0 +1,68 @@
+package app.model.Grid;
+
+import app.controller.linAlg.Vector;
+import app.model.agents.Cells.*;
+
+public class Grid
+{
+    protected int rowSize;
+    protected int colSize;
+    protected Cell[][] grid;
+    protected double cellSize = 1.0;
+    protected CellType type;
+
+
+    public Grid(double length, double width, CellType type)
+    {
+        rowSize = (int)(length / cellSize);
+        colSize = (int)(width / cellSize);
+        this.type = type;
+
+        grid = new Cell[rowSize][colSize];
+        initializeAllCells();
```

Comment:
This would be better if made functional
```java 
grid = initialiseAllCells()
```
where the method is returning a new `Cell[][]` grid.  This makes it easier to understand where parts of the class are being modified and reduces the chances of unintended side effects due to accidental calls